pkg/api/validate/dynamic/serviceprincipal_test.go

```diff
@@ -5,101 +5,79 @@ package dynamic
 
```

Comment:
I just have to comment that this file's changes is one of the most satisfying things I've seen in a while, it reads like a sigh of relief. Excellent.